src/bootstrap/src/core/config/flags.rs

```diff
@@ -701,3 +701,25 @@ pub fn get_completion(shell: &dyn Generator, path: &Path) -> Option<String> {
     }
     Some(String::from_utf8(buf).expect("completion script should be UTF-8"))
 }
+
+/// Return the help of the bootstrap if the result differs from the current
+/// help in `path`. It always returns the help if `path` does not exist.
+pub fn get_help(path: &Path) -> Option<String> {
```

Comment:
I think we should simplify this. Within the Rust part of bootstrap, it never makes sense to read from the file (except in the test that it's up to date, maybe). I don't think we need a wrapper for read_to_string, so I would make this argument-free (`fn top_level_help() -> String`).

Then GenerateHelp just writes the results, and the test compares the results or calls GenerateHelp if --bless is passed.